tidysic/parser/tree.py

```diff
@@ -25,8 +32,64 @@ def _parse(self) -> None:
         """
         for path in self._root.iterdir():
             if path.is_dir():
-                self.children.add(Tree(path))
+                child = Tree(path)
+                if child._contains_tags():
+                    self.children.add(child)
+                else:
+                    self.clutter_files.add(TaggedFile(path))
             elif AudioFile.is_audio_file(path):
                 self.audio_files.add(AudioFile(path))
             else:
-                self.clutter_files.add(path)
+                self.clutter_files.add(TaggedFile(path))
+
+        if self._contains_tags():
+            self._tag_clutter()
+
+    def _contains_tags(self) -> bool:
+        """
+        Returns True if this node or any of its children has audio files.
+        """
+        return (
+            self.common_tags is not None
+            or
+            len(self.audio_files) > 0
+            or
+            any(
+                child._contains_tags()
+                for child in self.children
+            )
+        )
+
+    def _tag_clutter(self):
+        """
+        Tags non-audio files with the tags common to all audio files in the same
+        directory and subdirectories.
+        """
+        common_tags = self._find_common_tags()
+        self._apply_tags_to_clutter(common_tags)
+
+    def _find_common_tags(self) -> Taggable:
```

Comment:
Picky but: if I'm not mistaken, this method will work only if `self._contains_tags() == True`. Since you know that, you are doing the check before calling it (e.g., line 45). Fairly easy for someone else to generate a runtime error though by forgetting this check.